filestore/s3.go

```diff
@@ -151,6 +151,7 @@ func (fs *S3FileStore) StoreFile(le *logrus.Entry, p *StoreFileParams) (out *Fil
 		return nil, errors.New("s3 store request: " + errstr)
 	}
 	defer resp.Body.Close()
+	defer io.Copy(io.Discard, resp.Body)
```

Comment:
The order in which you defer the functions in Go is critical because defer statements are executed in last-in, first-out order when the function returns.

Given the code:

go

defer resp.Body.Close()
defer io.Copy(io.Discard, resp.Body)

These statements will be executed in reverse order when the function returns, meaning io.Copy(io.Discard, resp.Body) will execute before resp.Body.Close().

Very cool!